ui/src/dashboards/apis/index.ts

```diff
@@ -1,102 +1,161 @@
+  const dashboards = await client.dashboards.getAll()
+
+  return dashboards.map(d => ({
+    ...d,
+    labels: d.labels.map(addLabelDefaults),
+    cells: addDashboardIDToCells(d.cells, d.id),
+  }))
 }
 
-export const createDashboard = async dashboard => {
-  try {
-    return await AJAX({
-      method: 'POST',
-      resource: 'dashboards',
-      data: dashboard,
-    })
-  } catch (error) {
-    console.error(error)
-    throw error
+export const getDashboard = async (id: string): Promise<Dashboard> => {
+  const dashboard = await client.dashboards.get(id)
+
+  return {
+    ...dashboard,
+    labels: dashboard.labels.map(addLabelDefaults),
+    cells: addDashboardIDToCells(dashboard.cells, dashboard.id),
   }
 }
 
-export const deleteDashboard = async dashboard => {
-  try {
-    return await AJAX({
-      method: 'DELETE',
-      url: dashboard.links.self,
-    })
-  } catch (error) {
-    console.error(error)
-    throw error
+export const createDashboard = async (
+  props: CreateDashboardRequest
+): Promise<Dashboard> => {
+  const dashboard = await client.dashboards.create(props)
+
+  return {
+    ...dashboard,
+    labels: dashboard.labels.map(addLabelDefaults),
+    cells: addDashboardIDToCells(dashboard.cells, dashboard.id),
   }
 }
 
-export const addDashboardCell = async (dashboard, cell) => {
-  try {
-    return await AJAX({
-      method: 'POST',
-      url: dashboard.links.cells,
-      data: cell,
-    })
-  } catch (error) {
-    console.error(error)
-    throw error
+export const deleteDashboard = async (dashboard: Dashboard): Promise<void> => {
+  await client.dashboards.delete(dashboard.id)
+}
+
+export const updateDashboard = async (
+  dashboard: Dashboard
+): Promise<Dashboard> => {
+  const updated = await client.dashboards.update(dashboard.id, dashboard)
+
+  return {
+    ...updated,
+    labels: updated.labels.map(addLabelDefaults),
+    cells: addDashboardIDToCells(updated.cells, updated.id),
   }
 }
 
-export const deleteDashboardCell = async cell => {
-  try {
-    return await AJAX({
-      method: 'DELETE',
-      url: cell.links.self,
+export const addCell = async (
+  dashboardID: string,
+  cell: NewCell
+): Promise<Cell> => {
+  const result = await client.dashboards.createCell(dashboardID, cell)
+
+  const cellWithID = {...result, dashboardID}
+
+  return cellWithID
+}
+
+export const updateCells = async (
+  id: string,
+  cells: Cell[]
+): Promise<Cell[]> => {
+  const result = await client.dashboards.updateAllCells(id, cells)
+
+  return addDashboardIDToCells(result, id)
+}
+
+export const deleteCell = async (
+  dashboardID: string,
+  cell: Cell
+): Promise<void> => {
+  await client.dashboards.deleteCell(dashboardID, cell.id)
+}
+
+export const addDashboardLabels = async (
+  dashboardID: string,
+  labels: ILabel[]
+): Promise<ILabel[]> => {
+  const addedLabels = await Promise.all(
+    labels.map(async label => {
+      return client.dashboards.addLabel(dashboardID, label.id)
```

Comment:
i believe there is already a `client.dashboards.addLabels` that takes in a list  of label ids and does this. how do you feel about using that function rather than reimplementing it here ?

Reply:
will do in next PR